src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -79,6 +79,36 @@ def customDID(pname):
     return pname
 
 
+def updateTransitionRecord(doc, userDN, logger):
+    """
+    Update transition record based on the following logic:
+    - check previous fraction value and add new transition record if it is
```

Comment:
This docstring looks confusing to me.
I think we discussed that a transition record will only depend on a difference between the container fraction and the last transition fraction, right?

Reply:
yes, after iteration we concluded that second use-case is not relevant. Correcting the docstring.

Reply:
In addition, I would also rename this method, as it is actually not updating a record, but creating one.

If I may suggest, a docstring like "Create a new transition record if the current container fraction (containerFraction) differs from the last transition record container fraction (transition.containerFraction). If they are the same, nothing needs to be done with the transition records." looks more clear to me, keeping the parameters and return documentation, of course.